external/fv3fit/tests/test_bptt.py

```diff
@@ -0,0 +1,272 @@
+        dt.total_seconds()
+    )
+
+    np.testing.assert_allclose(
+        Q1_train_tendency - dQ1[:, 1, :],
+        train_dataset["air_temperature_tendency_due_to_model"][:, 1, :].values,
+        atol=1e-6,
+    )
+    np.testing.assert_allclose(
+        Q2_train_tendency - dQ2[:, 1, :],
+        train_dataset["specific_humidity_tendency_due_to_model"][:, 1, :].values,
+        atol=1e-6,
+    )
+
+
+def test_predict_model_gives_different_tendencies(train_dataset, sample_dim_name, dt):
+    model = BPTTModel(
+        sample_dim_name,
+        ["a", "b"],
+        n_units=32,
+        n_hidden_layers=4,
+        kernel_regularizer=None,
+        train_batch_size=48,
+        optimizer="adam",
+    )
+    model.build_for(train_dataset)
+
+    tendency_ds = model.predictor_model.predict(train_dataset.isel(time=0))
+    assert not np.all(tendency_ds["dQ1"].values == tendency_ds["dQ2"].values)
+
+
+def test_train_model_gives_different_outputs(train_dataset, sample_dim_name):
+    model = BPTTModel(
+        sample_dim_name,
+        ["a", "b"],
+        n_units=32,
+        n_hidden_layers=4,
+        kernel_regularizer=None,
+        train_batch_size=48,
+        optimizer="adam",
+    )
+    model.build_for(train_dataset)
+
+    air_temperature, specific_humidity = model.train_keras_model.predict(
+        x=model.get_keras_inputs(train_dataset)
+    )
+    assert not np.all(air_temperature == specific_humidity)
+
+
+def test_integrate_stepwise(train_dataset, sample_dim_name):
+    np.random.seed(0)
+    tf.random.set_seed(1)
+    model = BPTTModel(
+        sample_dim_name,
+        ["a", "b"],
+        n_units=32,
+        n_hidden_layers=4,
+        kernel_regularizer=None,
+        train_batch_size=48,
+        optimizer="adam",
+    )
+    model.build_for(train_dataset)
+
+    air_temperature, specific_humidity = model.train_keras_model.predict(
+        x=model.get_keras_inputs(train_dataset)
+    )
+    out_ds = model.predictor_model.integrate_stepwise(train_dataset)
+    np.testing.assert_allclose(
+        out_ds["air_temperature"].values[:, 0, :], air_temperature[:, 0, :], atol=1e-6
+    )
+    np.testing.assert_allclose(
+        out_ds["specific_humidity"].values[:, 0, :],
+        specific_humidity[:, 0, :],
+        atol=1e-6,
+    )
+    np.testing.assert_allclose(
+        out_ds["air_temperature"].values[:, 1, :], air_temperature[:, 1, :], atol=1e-6
+    )
+    np.testing.assert_allclose(
+        out_ds["specific_humidity"].values[:, 1, :],
+        specific_humidity[:, 1, :],
+        atol=1e-6,
+    )
+    np.testing.assert_allclose(
+        out_ds["air_temperature"].values, air_temperature, atol=1e-5
+    )
+    np.testing.assert_allclose(
+        out_ds["specific_humidity"].values, specific_humidity, atol=1e-5
+    )
+    assert not np.all(air_temperature == specific_humidity)
+
+
+def test_reloaded_model_gives_same_outputs(train_dataset, sample_dim_name):
+    model = BPTTModel(
```

Comment:
I'd like to skip this for prioritization reasons if that's OK. At a glance I don't see much functionality to merge because of minor differences in the API, so this could take a fair bit of time.